drivers/serial/uart_ns16550.c

```diff
@@ -190,14 +190,22 @@
 
 #define IIRC(dev) (DEV_DATA(dev)->iir_cache)
 
+#ifdef CONFIG_NS16550_REG_SHIFT
```

Comment:
lets not use CONFIG_ here, since we don't have a Kconfig symbol we are mapping onto.

How about: DT_NS16550_REG_SHIFT